lib/Twig/Template.php

```diff
@@ -38,9 +38,19 @@
     protected $blocks = array();
     protected $traits = array();
 
+    /**
+     * @internal
+     */
+    protected $extensions = array();
+
     public function __construct(Twig_Environment $env)
     {
         $this->env = $env;
+        foreach ($env->getExtensions() as $extension) {
```

Comment:
Would it make sense to cache this in the Env instead? Because now it's done for every template load, and it doesn't change between loading right?

Reply:
And it's cleaner to do this logic only once at the right place. See #2649